[FEATURE] Add record list download presets (#529)
TYPO3-Documentation/TYPO3CMS-Reference-TSconfig: Documentation/PageTsconfig/Mod/WebList.rst

    mod.web_list.displayColumnSelector = 0

..  _pageTsConfigWebList-downloadPresets:

downloadPresets
===============

..  confval:: downloadPresets.[table]
    :name: mod-web-list-downloadPresets
    :type: array of presets

    ..  versionadded:: 13.2

>         A new feature has been introduced, adding the ability to pick the exported
>         data columns from a list of configurable presets.

Reviewer: The `exported data columns` are the already exported columns, but I think you mean the data columns which has to be exported:

```suggestion
        A new function has been introduced that makes it possible to select the data columns
        to be exported from a list of configurable presets.
```

>     This property adds presets of preselected fields to the download area in

Reviewer: In this case I think `fields` is OK, as the visible content of the fields of the TCEform will be used for the export, and not the plain value from database table.

    the :guilabel:`Web > List` backend module.

    Those presets can be configured via page TSconfig, and can also be
    overridden via user TSconfig (for example, to expand certain presets
    only to specific users).

    Each entry of :typoscript:`mod.web_list.downloadPresets`
    defines the table name on the first level, followed by
    any number of presets.

    Each preset contains a :typoscript:`label` (the displayed name of the preset,
    which can be a locallang key), a comma-separated list of each column that
    should be included in the export as :typoscript:`columns` and optionally

>     a :typoscript:`identifier`. In case :typoscript:`identifier` is not provided,

Reviewer: ```suggestion
    an :typoscript:`identifier`. In case :typoscript:`identifier` is not provided,
```

    the identifier is generated as hash of the :typoscript:`label` and
    :typoscript:`columns`.

    Since any table can be configured for a preset, any extension
    can deliver a defined set of presets through the
    :file:`EXT:my_extension/Configuration/page.tsconfig` file and
    their table name(s).

    ..  todo:: Link event after it is merged

>     Additionally, the list of presets can be manipulated via the event

Reviewer 2: ```suggestion
    Additionally, the list of presets can be manipulated via the PSR-14 event
```

>     `BeforeRecordDownloadPresetsAreDisplayedEvent`.

Reviewer 2: ```suggestion
    :php:`BeforeRecordDownloadPresetsAreDisplayedEvent`.
```

..  _pageTsConfigWebList-downloadPresets-example:

Example: Create download presets for table page
-----------------------------------------------

..  literalinclude:: _WebList/_downloadPresets.tsconfig
    :caption: EXT:my_extension/Configuration/page.tsconfig

> This can be manipulated with user TSConfig by adding the :typoscript:`page.`

Reviewer: Thank you for sharing the link to our glossary ;-) 

```suggestion
This can be manipulated with user TSconfig by adding the :typoscript:`page.`
```

Author: Copy and pasted this from the changelog entry. You want to fix it there?

> prefix. User TSConfig is loaded after page TSConfig, so you can overwrite

Reviewer: ```suggestion
prefix. User TSconfig is loaded after page TSconfig, so you can overwrite
```

> existing keys or replace the whole list of keys:

Reviewer: I know what you mean with `keys`, but until now the word `keys` was not used. What about: "You can overwrite the existing default settings using the same TypoScript path."

Author: Done, Good idea.

> existing keys or replace the whole list of keys:

Author: ```suggestion
the existing default settings using the same TypoScript path.
```
